Sources/SwiftFusion/MCMC/sample.swift

```diff
@@ -0,0 +1,39 @@
+// Copyright 2020 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Implements Markov chain Monte Carlo via repeated TransitionKernel steps
+/// Inspired by tfp [sample_chain](https://www.tensorflow.org/probability/api_docs/python/tfp/mcmc/sample_chain)
+public func sampleChain<State, Kernel:TransitionKernel>(_ num_results:Int,
+                                                        _ init_state:State,
+                                                        _ kernel:Kernel,
+                                                        _ num_burnin_steps:Int) -> Array<State>
+where Kernel.State==State {
```

Comment:
This should not be necessary along with the previous `<State,` parameter as you can always use `Kernel.State` directly.

Reply:
Good point! Done